File: core/src/com/nopalsoft/dragracer/handlers/GameServicesHandler.java
package com.nopalsoft.dragracer.handlers;

public interface GameServicesHandler {

    // This method abstracts away GPGS or AGC.
    void submitScore(long score);

    // This method abstracts away GPGS or AGC.
    void getLeaderboard();

    boolean isSignedIn();

    void signIn();

}

File: core/src/com/nopalsoft/dragracer/handlers/GameServicesHandler.kt
package com.nopalsoft.dragracer.handlers

interface GameServicesHandler {

    // This method abstracts away GPGS or AGC.
    fun submitScore(score:Long)

    // This method abstracts away GPGS or AGC.
    fun getLeaderboard()

    fun isSignedIn():Boolean

    fun signIn()
}